snapshot.go

```diff
@@ -0,0 +1,479 @@
+package frostdb
```

Comment:
Agreed. I was looking at doing this, but I think came to the conclusion that we needed access to some db internals that would make the API a bit murky. Can take another look at doing this.

Reply:
I tried creating the snapshots package and a lot of internal dependencies can be exported, however when loading a snapshots, the code manipulates a table's active block index. It feels wrong to export this behavior and moving just the index code to the `frostdb` package doesn't seem like a clean break. What do you think?

Reply:
I can maybe try to return the new index from the snapshot code to be loaded. Not sure I like it but will work on a commit so that we can see the final result.

Reply:
Hmm, but since we can be loading multiple tables we'd have to return some way to map table names to indexes. I think it's probably not worth doing so at this point...